torch/_inductor/codegen/wrapper.py

```diff
@@ -282,19 +282,14 @@ def codegen(
                 else:
                     code.writeline(f"device_guard.set_index({self.device_idx});")
         else:
-            # Note _DeviceGuard has less overhead than device, but only accepts
-            # integers
-            code.writeline(f"with {V.graph.device_ops.device_guard(self.device_idx)}:")
-            device_cm_stack.enter_context(code.indent())
             code.writeline(V.graph.device_ops.set_device(self.device_idx))
```

Comment:
Don't we need to restore the original active device when exiting the wrapper code? Both for normal exits and on exceptions.

Reply:
Good point. Let me look further into this.